pennylane_lightning/src/gates/cpu_kernels/GateImplementationsLM.hpp

```diff
@@ -1355,4 +1357,294 @@ class GateImplementationsLM : public PauliGenerator<GateImplementationsLM> {
+                                             const std::vector<size_t> &, bool)
+    -> double;
+extern template auto GateImplementationsLM::applyGeneratorIsingZZ(
+    std::complex<float> *, size_t, const std::vector<size_t> &, bool) -> float;
+
+extern template auto GateImplementationsLM::applyGeneratorCRX(
+    std::complex<float> *, size_t, const std::vector<size_t> &, bool) -> float;
+extern template auto
+GateImplementationsLM::applyGeneratorCRX(std::complex<double> *, size_t,
+                                         const std::vector<size_t> &, bool)
+    -> double;
+
+extern template auto GateImplementationsLM::applyGeneratorCRY(
+    std::complex<float> *, size_t, const std::vector<size_t> &, bool) -> float;
+extern template auto
+GateImplementationsLM::applyGeneratorCRY(std::complex<double> *, size_t,
+                                         const std::vector<size_t> &, bool)
+    -> double;
+
+extern template auto GateImplementationsLM::applyGeneratorCRZ(
+    std::complex<float> *, size_t, const std::vector<size_t> &, bool) -> float;
+extern template auto
+GateImplementationsLM::applyGeneratorCRZ(std::complex<double> *, size_t,
+                                         const std::vector<size_t> &, bool)
+    -> double;
+
+extern template auto GateImplementationsLM::applyGeneratorControlledPhaseShift(
+    std::complex<double> *, size_t, const std::vector<size_t> &, bool)
+    -> double;
+extern template auto GateImplementationsLM::applyGeneratorControlledPhaseShift(
+    std::complex<float> *, size_t, const std::vector<size_t> &, bool) -> float;
+
+extern template auto GateImplementationsLM::applyGeneratorSingleExcitation(
+    std::complex<double> *, size_t, const std::vector<size_t> &, bool)
+    -> double;
+extern template auto GateImplementationsLM::applyGeneratorSingleExcitation(
+    std::complex<float> *, size_t, const std::vector<size_t> &, bool) -> float;
+
+extern template auto GateImplementationsLM::applyGeneratorSingleExcitationMinus(
+    std::complex<double> *, size_t, const std::vector<size_t> &, bool)
+    -> double;
+extern template auto GateImplementationsLM::applyGeneratorSingleExcitationMinus(
+    std::complex<float> *, size_t, const std::vector<size_t> &, bool) -> float;
+
+extern template auto GateImplementationsLM::applyGeneratorSingleExcitationPlus(
+    std::complex<double> *, size_t, const std::vector<size_t> &, bool)
+    -> double;
+extern template auto GateImplementationsLM::applyGeneratorSingleExcitationPlus(
+    std::complex<float> *, size_t, const std::vector<size_t> &, bool) -> float;
+
+extern template auto
+GateImplementationsLM::applyGeneratorMultiRZ(std::complex<double> *, size_t,
+                                             const std::vector<size_t> &, bool)
+    -> double;
+extern template auto GateImplementationsLM::applyGeneratorMultiRZ(
+    std::complex<float> *, size_t, const std::vector<size_t> &, bool) -> float;
```

Comment:
Given the size of this file, should we maybe move the above off to a separate file to be included?

Reply:
We can also later see if there is a way to use a parameter pack to forward declare both the `float` and `double` templates to reduce the length.